pkg/agent/loadbalancer/servers.go

```diff
@@ -3,14 +3,48 @@ package loadbalancer
 import (
 	"context"
 	"errors"
+	"fmt"
 	"math/rand"
 	"net"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/k3s-io/k3s/pkg/version"
+	http_dialer "github.com/mwitkow/go-http-dialer"
+	"golang.org/x/net/proxy"
 
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-var defaultDialer = &net.Dialer{}
+var defaultDialer proxy.Dialer = &net.Dialer{}
```

Comment:
making this a `proxy.Dialer` feels wrong, but since https://github.com/golang/go/issues/9360 is unresolved and core golang doesn't have a dialer interface, I guess this is OK?

Reply:
You tell me, it may feel wrong but is working (surprisingly), this is to avoid using 2 variables with 2 different types, enforcing proxy.Dialer seems to be working just fine both when proxy is in use and when it's not.

Reply:
yeah, like I said I think it's OK - it just looks odd.